src/providers/idp/idp_id.c

```diff
@@ -0,0 +1,675 @@
+}
+
+static void idp_type_get_done(struct tevent_req *subreq)
+{
+    struct idp_type_get_state *state;
+    struct tevent_req *req;
+    errno_t ret;
+
+    uint8_t *buf;
+    ssize_t buflen;
+
+    req = tevent_req_callback_data(subreq, struct tevent_req);
+    state = tevent_req_data(req, struct idp_type_get_state);
+
+    ret = handle_oidc_child_recv(subreq, state, &buf, &buflen);
+    talloc_zfree(subreq);
+    if (ret != EOK) {
+        state->dp_error = DP_ERR_FATAL;
+        tevent_req_error(req, ret);
+        return;
+    }
+
+    DEBUG(SSSDBG_TRACE_ALL, "[%zd][%.*s]\n", buflen, (int) buflen, buf);
+    switch (state->lookup_type) {
+    case IDP_LOOKUP_USER:
+        ret = eval_user_buf(state->idp_id_ctx, NULL, buf, buflen);
+        break;
+    case IDP_LOOKUP_GROUP:
+        ret = eval_group_buf(state->idp_id_ctx, NULL, buf, buflen);
+        if (ret == EOK) {
+            DEBUG(SSSDBG_TRACE_ALL, "Looking up group members.\n");
+
+            state->lookup_type = IDP_LOOKUP_GROUP_MEMBERS;
+            talloc_zfree(state->idp_req);
+
+            ret = idp_type_get_step(req);
+            if (ret == EOK) {
+                return;
+            }
+            DEBUG(SSSDBG_OP_FAILURE, "Failed to lookup group members.\n");
+        }
+        break;
+    case IDP_LOOKUP_GROUP_MEMBERS:
+        ret = eval_user_buf(state->idp_id_ctx, state->filter_value,
+                            buf, buflen);
+        break;
+    case IDP_LOOKUP_USER_GROUPS:
+        ret = eval_group_buf(state->idp_id_ctx, state->filter_value,
+                             buf, buflen);
+        break;
+    default:
+        DEBUG(SSSDBG_OP_FAILURE, "Unsupported lookup type [%d].\n",
+                                 state->lookup_type);
+        ret = EINVAL;
+    }
+    if (ret != EOK) {
+        DEBUG(SSSDBG_OP_FAILURE,
+              "Failed to evaluate user data returned by oidc_child.\n");
+        state->dp_error = DP_ERR_FATAL;
+        tevent_req_error(req, ret);
+        return;
+    }
+
+    state->dp_error = DP_ERR_OK;
+    tevent_req_done(req);
+}
+
+
+int idp_type_get_recv(struct tevent_req *req, int *dp_error_out, int *idp_ret)
+{
+    struct idp_type_get_state *state;
+
+    state = tevent_req_data(req, struct idp_type_get_state);
+
+    if (dp_error_out != NULL) {
+        *dp_error_out = state->dp_error;
+    }
+
+    if (idp_ret != NULL) {
+        *idp_ret = state->idp_ret;
+    }
+
+    TEVENT_REQ_RETURN_ON_ERROR(req);
+
+    return EOK;
+}
+
+struct tevent_req *idp_users_get_send(TALLOC_CTX *memctx,
+                                      struct tevent_context *ev,
+                                      struct idp_id_ctx *idp_id_ctx,
+                                      const char *filter_value,
+                                      int filter_type,
+                                      const char *extra_value,
+                                      bool noexist_delete,
+                                      bool set_non_posix)
```

Comment:
What is the purpose of this `set_non_posix` arg?

Reply:
I added this to have a comparable argument list to the corresponding functions of the other providers. In general the IdP provider can be extended to support application (non-POSIX) domains, that's why I kept it.

Reply:
Could you please add a comment it's currently not used and a potential future plumbing point?